cli/tests/pcluster/configure/test_subnet_cidr.py

```diff
@@ -0,0 +1,79 @@
+from assertpy import assert_that
+from pcluster.utils import decide_cidr, get_subnet_cidr
+
+
+def test_empty_vpc():
+    assert_that(decide_cidr(vpc_cidr="10.0.0.0/16", occupied_cidr=[], target_size=250)).is_equal_to("10.0.0.0/24")
+    assert_that(decide_cidr(vpc_cidr="10.0.0.0/8", occupied_cidr=[], target_size=250)).is_equal_to("10.0.0.0/24")
+    assert_that(decide_cidr(vpc_cidr="10.2.0.0/16", occupied_cidr=[], target_size=250)).is_equal_to("10.2.0.0/24")
+    assert_that(decide_cidr(vpc_cidr="10.2.0.0/25", occupied_cidr=[], target_size=500)).is_none()
+    assert_that(decide_cidr(vpc_cidr="10.2.0.0/25", occupied_cidr=[], target_size=100)).is_equal_to("10.2.0.0/25")
+
+
+def test_no_space_vpc():
+    assert_that(decide_cidr(vpc_cidr="10.0.0.0/16", occupied_cidr=["10.0.0.0/16"], target_size=1)).is_none()
+    assert_that(
+        decide_cidr(vpc_cidr="10.0.0.0/16", occupied_cidr=["10.0.0.0/17", "10.0.128.0/17"], target_size=1)
+    ).is_none()
+    assert_that(
+        decide_cidr(
+            vpc_cidr="10.0.0.0/16", occupied_cidr=["10.0.0.0/18", "10.0.64.0/18", "10.0.128.0/18"], target_size=16385
+        )
+    ).is_none()
+    assert_that(
+        decide_cidr(vpc_cidr="10.0.0.0/16", occupied_cidr=["10.0.0.0/18", "10.0.128.0/17"], target_size=16385)
+    ).is_none()
+
+
+def test_space_needed_bigger_than_allocated_subnets():
```

Comment:
Please add a comment to the function to describe we are testing the "promote" feature.